std/digest/crc.d

```diff
@@ -169,36 +197,46 @@ struct CRC32
                     enum hasLittleEndianUnalignedReads = false; // leave decision to optimizer
                 static if (hasLittleEndianUnalignedReads)
                 {
-                    uint one = (cast(uint*) data.ptr)[0] ^ crc;
+                    uint one = (cast(uint*) data.ptr)[0];
                     uint two = (cast(uint*) data.ptr)[1];
                 }
                 else
                 {
-                    uint one = (data.ptr[3] << 24 | data.ptr[2] << 16 | data.ptr[1] << 8 | data.ptr[0]) ^ crc;
+                    uint one = (data.ptr[3] << 24 | data.ptr[2] << 16 | data.ptr[1] << 8 | data.ptr[0]);
                     uint two = (data.ptr[7] << 24 | data.ptr[6] << 16 | data.ptr[5] << 8 | data.ptr[4]);
                 }
 
+                static if (N == 32)
+                {
+                    one ^= crc;
+                }
+                else
+                {
+                    one ^= (crc & 0xffffffff);
+                    two ^= (crc >> 32);
+                }
+
                 crc =
-                    crc32Tables[0][two >> 24] ^
-                    crc32Tables[1][(two >> 16) & 0xFF] ^
-                    crc32Tables[2][(two >>  8) & 0xFF] ^
-                    crc32Tables[3][two & 0xFF] ^
-                    crc32Tables[4][one >> 24] ^
-                    crc32Tables[5][(one >> 16) & 0xFF] ^
-                    crc32Tables[6][(one >>  8) & 0xFF] ^
-                    crc32Tables[7][one & 0xFF];
+                    tables[0][two >> 24] ^
+                    tables[1][(two >> 16) & 0xFF] ^
+                    tables[2][(two >>  8) & 0xFF] ^
+                    tables[3][two & 0xFF] ^
+                    tables[4][one >> 24] ^
+                    tables[5][(one >> 16) & 0xFF] ^
+                    tables[6][(one >>  8) & 0xFF] ^
+                    tables[7][one & 0xFF];
 
                 data = data[8 .. $];
             }
             // remaining 1 to 7 bytes
             foreach (d; data)
-                crc = (crc >> 8) ^ crc32Tables[0][(crc & 0xFF) ^ d];
+                crc = (crc >> 8) ^ tables[0][(crc & 0xFF) ^ d];
             _state = crc;
         }
         ///
         unittest
         {
-            CRC32 dig;
+            CRC dig;
```

Comment:
That's the reason why CircleCi fails with:

```d
out/std_digest_crc.d(46): Error: struct std.digest.crc.CRC(uint N, ulong P) if (N == 32 || N == 64) is used as a type
out/std_digest_crc.d(59): Error: struct std.digest.crc.CRC(uint N, ulong P) if (N == 32 || N == 64) is used as a type
out/std_digest_crc.d(71): Error: struct std.digest.crc.CRC(uint N, ulong P) if (N == 32 || N == 64) is used as a type
out/std_digest_crc.d(73): Error: undefined identifier 'R'
```

The script at CircleCi seperates all public unittests and tries to run them with the newly compiled release library to ensure that all unittests on dlang.org are runnable as they are.

Btw I remember there was a long discussion last summer about the problems of using `unittest` blocks in templated code - in short in theory it's discouraged at Phobos because the `unittest` block gets instantiated very every new instance.